modulus/utils/generative/stochastic_sampler.py

```diff
@@ -82,68 +82,70 @@ def image_batching(
+    ).to(input.device)
     image_padding = torch.nn.ReflectionPad2d(
         (boundary_pix, pad_x_right, boundary_pix, pad_y_right)
-    ).cuda()  # (padding_left,padding_right,padding_top,padding_bottom)
+    ).to(
+        input.device
+    )  # (padding_left,padding_right,padding_top,padding_bottom)
     input_padded = image_padding(input)
     patch_num = patch_num_x * patch_num_y
     if input_interp is not None:
         output = torch.zeros(
             patch_num * batch_size,
             input.shape[1] + input_interp.shape[1],
-            patch_shape_x,
             patch_shape_y,
-        ).cuda()
+            patch_shape_x,
+        ).to(input.device)
     else:
         output = torch.zeros(
-            patch_num * batch_size, input.shape[1], patch_shape_x, patch_shape_y
-        ).cuda()
+            patch_num * batch_size, input.shape[1], patch_shape_y, patch_shape_x
+        ).to(input.device)
     for x_index in range(patch_num_x):
         for y_index in range(patch_num_y):
             x_start = x_index * (patch_shape_x - overlap_pix - boundary_pix)
             y_start = y_index * (patch_shape_y - overlap_pix - boundary_pix)
             if input_interp is not None:
                 output[
-                    (x_index * patch_num_x + y_index)
-                    * batch_size : (x_index * patch_num_x + y_index + 1)
+                    (x_index * patch_num_y + y_index)
+                    * batch_size : (x_index * patch_num_y + y_index + 1)
                     * batch_size,
                 ] = torch.cat(
                     (
                         input_padded[
                             :,
                             :,
-                            x_start : x_start + patch_shape_x,
                             y_start : y_start + patch_shape_y,
+                            x_start : x_start + patch_shape_x,
                         ],
                         input_interp,
                     ),
                     dim=1,
                 )
             else:
                 output[
-                    (x_index * patch_num_x + y_index)
-                    * batch_size : (x_index * patch_num_x + y_index + 1)
+                    (x_index * patch_num_y + y_index)
+                    * batch_size : (x_index * patch_num_y + y_index + 1)
                     * batch_size,
                 ] = input_padded[
                     :,
                     :,
-                    x_start : x_start + patch_shape_x,
                     y_start : y_start + patch_shape_y,
+                    x_start : x_start + patch_shape_x,
                 ]
     return output
 
 
 def image_fuse(
-    input: Tensor,
-    img_shape_x: int,
-    img_shape_y: int,
-    patch_shape_x: int,
-    patch_shape_y: int,
-    batch_size: int,
-    overlap_pix: int,
-    boundary_pix: int,
+    input,
+    img_shape_y,
+    img_shape_x,
+    patch_shape_y,
+    patch_shape_x,
+    batch_size,
+    overlap_pix,
+    boundary_pix,
```

Comment:
Can we add the type hints back?

Reply:
Yes for sure. Sorry for delay in response, the notification went to my personal email